kvclientpool.go

```diff
@@ -394,6 +418,13 @@ func (p *kvClientPool) GetClient(ctx context.Context) (KvClient, error) {
 }
 
 func (p *kvClientPool) getClientSlow(ctx context.Context) (KvClient, error) {
+	// Check if we're already closed and if so return an error.
+	select {
+	case <-p.closed:
```

Comment:
There are some weird details about dealing with channels, and there is a performance impact to doing a select like this.  It would likely be better to have a `closed bool` protected by the lock, and then the closedSig is only used to watch the channel if we are going to be blocking for another signal.